pkg/pillar/scripts/device-steps.sh

```diff
@@ -17,7 +17,7 @@ LISPDIR=/opt/zededa/lisp
 LOGDIRA=$PERSISTDIR/IMGA/log
 LOGDIRB=$PERSISTDIR/IMGB/log
 AGENTS0="logmanager ledmanager nim nodeagent"
-AGENTS1="zedmanager zedrouter domainmgr downloader verifier identitymgr zedagent lisp-ztr baseosmgr wstunnelclient"
+AGENTS1="upgradeconverter zedmanager zedrouter domainmgr downloader verifier identitymgr zedagent lisp-ztr baseosmgr wstunnelclient"
```

Comment:
You need to run this before ledmamager hence it can not be listed in AGENTS1.
Instead you need explicit code in device-steps.sh before ledmanager is run, and presumably this needs to pass the correct arguments to upgradeconverter

Reply:
@archishou - I will take it from here. I will fix the device steps and run some tests on a real device before merging it up..

Reply:
Ok. I've cleared up all the other comments and all tests still pass.

Reply:
As I pointed out above, with the dis-aggregation effort now underway -- the right place for upgradeconverter is now in storage-init container.

This also raises an important issue for me in that we need to make sure that this "upgrade" code only runs once per version of the script ('cuz otherwise all heck has a tendency to break lose). One option is to introduce the notion of a "version" of /persist and only trigger this upgrade code if persist.version < eve.version

Reply:
@kalyan-nidumolu Did you change device-steps to actually call upgradeconverter to do some conversion? Otherwise testing isn't very useful.

Reply:
@rvs Do know how a persist-version would work when we can fallback or downgrade to the old until we commit (and have upgradeconverter remove the old files). During the intermediate we have two versions in /persist/